src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -5172,33 +5172,11 @@ void Unit::SetPowerType(Powers new_powertype)
             ToPlayer()->SetGroupUpdateFlag(GROUP_UPDATE_FLAG_POWER_TYPE);
     }
     /*else if (Pet* pet = ToCreature()->ToPet()) TODO 6.x
-    {
         if (pet->isControlled())
-            pet->SetGroupUpdateFlag(GROUP_UPDATE_FLAG_PET_POWER_TYPE);
-    }*/
-
-    float powerMultiplier = 1.0f;
-    if (!IsPet())
-        if (Creature* creature = ToCreature())
-            powerMultiplier = creature->GetCreatureTemplate()->ModMana;
-
-    switch (new_powertype)
-    {
-        default:
-        case POWER_MANA:
-            break;
-        case POWER_RAGE:
-            SetMaxPower(POWER_RAGE, uint32(std::ceil(GetCreatePowers(POWER_RAGE) * powerMultiplier)));
-            SetPower(POWER_RAGE, 0);
-            break;
-        case POWER_FOCUS:
-            SetMaxPower(POWER_FOCUS, uint32(std::ceil(GetCreatePowers(POWER_FOCUS) * powerMultiplier)));
-            SetPower(POWER_FOCUS, uint32(std::ceil(GetCreatePowers(POWER_FOCUS) * powerMultiplier)));
-            break;
-        case POWER_ENERGY:
-            SetMaxPower(POWER_ENERGY, uint32(std::ceil(GetCreatePowers(POWER_ENERGY) * powerMultiplier)));
-            break;
-    }
+            pet->SetGroupUpdateFlag(GROUP_UPDATE_FLAG_PET_POWER_TYPE);*/
+    
+    UpdateMaxPower(new_powertype);
+    SetPower(new_powertype, 0);
```

Comment:
does this cause you will lose all your mana on power type change, like druid shapeshift?

Reply:
Hmm, true, moonkin keeps the same mana I think. But other forms do reset power to 0 (energy, rage), no? I could extend arguments here: `void Unit::SetPowerType(Powers new_powertype, bool resetCurrentPower = true)`

Reply:
if power is the same dont reset maybe ?

Reply:
just restore the old way